src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/ODataResourceDeserializer.cs

```diff
@@ -331,7 +361,23 @@ public virtual object CreateResourceInstance(IEdmStructuredTypeReference structu
                     }
                     else
                     {
-                        ApplyResourceSetInNestedProperty(edmProperty, resource, resourceSetWrapper, readContext);
+                        ODataResourceSetWrapper resSetWrapper = resourceSetWrapper;
+                        IList<ODataResourceWrapper> resourceWrappers  = resourceSetWrapper.Resources.ToList();
+                        IList<Uri> resourceIds = new List<Uri>();
+                        foreach (ODataResourceWrapper resWrapper in resourceWrappers)
```

Comment:
```suggestion
                        foreach (ODataResourceWrapper resWrapper in resourceSetWrapper.Resources)
```
I don't think there's any benefit in introducing the `resourceWrappers` variable two lines above this one